### PR TITLE
feature(alternator): using docker dns server

### PR DESCRIPTION
**Pull request by fruch**
Using a docker that run with each loader command that act as a dns resolver
that is aware of the live nodes via the /localnodes api

### CODE REVIEW

**Comment by fruch** [comment]
leaving a run with this:
https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/fruch/job/longevity/job/longevity-alternator-3h-test/23/

File: unit_tests/test_cluster.py [hunk 1]
```diff
@@ -18,6 +18,10 @@
 class DummyNode(BaseNode):  # pylint: disable=abstract-method
     _database_log = None
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.remoter.stop()
+
```
Comment (nyh) [inline unit_tests/test_cluster.py]: I'm completely unfamiliar with this code, so I don't know what I'm reviewing, but how is this change related to a DNS server?

File: jenkins-pipelines/longevity-alternator-3h.jenkinsfile [hunk 2]
```diff
@@ -12,5 +12,5 @@ longevityPipeline(
     test_config: 'test-cases/longevity/longevity-alternator-3h.yaml',
 
     timeout: [time: 300, unit: 'MINUTES'],
-    email_recipients: '["qa@scylladb.com", "alternator@scylladb.com"]'
+    email_recipients: '["qa@scylladb.com","alternator@scylladb.com"]'
```
Comment (nyh) [inline jenkins-pipelines/longevity-alternator-3h.jenkinsfile]: ditto, unrelated to dns server :-)
But I don't know what your patch process is. Maybe you consider it fine to mix unrelated patches into one series like this.

File: docker/alternator-dns/dns_server.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+            pass
+        time.sleep(1)
+
+
+_thread.start_new_thread(livenodes_update, ())
+
+
+class Resolver:
+    def __init__(self):
+        self.address = '8.8.8.8'
```
Comment (nyh) [inline docker/alternator-dns/dns_server.py]: Nice. I see you added this feature I was meaning to add.

File: unit_tests/test_cluster.py (hunk 1 above)
Comment (fruch) [reply unit_tests/test_cluster.py]: this is not related, and I'll remove it, since it seem to not even solve the problem I was trying to fix.

File: jenkins-pipelines/longevity-alternator-3h.jenkinsfile (hunk 2 above)
Comment (fruch) [reply jenkins-pipelines/longevity-alternator-3h.jenkinsfile]: not related directly, but was annoying me while testing this.

File: docker/alternator-dns/dns_server.py (hunk 3 above)
Comment (fruch) [reply docker/alternator-dns/dns_server.py]: I did, but the thing is it's still not working.
i.e. the dns server is working, and I can get the name `alternator` resolved.

but the damn java/YCSB doesn't seem to honor the `networkaddress.cache.ttl=4` I've set in `$JAVA_HOME/jre/lib/security/java.security`, and doesn't query the dns again.

**Comment by bentsi** [comment]
@fruch did you build the new YCSB image and upload to ECR?

**Comment by fruch** [comment]
> @fruch did you build the new YCSB image and upload to ECR?

Nope, only to dockerhub
